From d35a3756e31016d531cfea05f1d3ece3e99a7821 Mon Sep 17 00:00:00 2001
From: Ash Charles <ashcharles@gmail.com>
Date: Fri, 18 Dec 2015 14:51:49 -0800
Subject: [PATCH] tfp410: RGB888 not RGB565 color-mapping for botball

Botball wires 24-bit color to the TFP410 HDMI encoder---the driver
should reflect this.  Normally, we'd just lose a little colo depth but
because of the AM335x uses a strange pin-mapping for RGB888 compared
with RGB565, we actually end up partially switching the blue and red
channels if the color-depth is set wrongly.

This is should be fixed properly by making the 'bpp' parameter settable
from the device-tree for the tfp410.

Signed-off-by: Ash Charles <ashcharles@gmail.com>
---
 drivers/gpu/drm/tilcdc/tilcdc_tfp410.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/tilcdc/tilcdc_tfp410.c b/drivers/gpu/drm/tilcdc/tilcdc_tfp410.c
index 094fc128e490..ac2065d71880 100644
--- a/drivers/gpu/drm/tilcdc/tilcdc_tfp410.c
+++ b/drivers/gpu/drm/tilcdc/tilcdc_tfp410.c
@@ -35,7 +35,7 @@ static const struct tilcdc_panel_info dvi_info = {
 		.ac_bias                = 255,
 		.ac_bias_intrpt         = 0,
 		.dma_burst_sz           = 16,
-		.bpp                    = 16,
+		.bpp                    = 24,
 		.fdd                    = 0x80,
 		.tft_alt_mode           = 0,
 		.sync_edge              = 0,
-- 
2.1.4

